File: addons/pos_sale_loyalty/models/loyalty_rule.py
# Part of Odoo. See LICENSE file for full copyright and licensing details.

from odoo import models


class LoyaltyRule(models.Model):
    _inherit = 'loyalty.rule'

    def _counts_for_points(self, line):
        # Lines settling a sales order already earned their points on the S.O.
        return not line.sale_order_origin_id and super()._counts_for_points(line)

File: addons/pos_sale_loyalty/static/src/overrides/models/loyalty_rule.js
import { LoyaltyRule } from "@pos_loyalty/app/models/loyalty_rule";
import { patch } from "@web/core/utils/patch";

patch(LoyaltyRule.prototype, {
    _countsForPoints(line) {
        return !line.sale_order_origin_id && super._countsForPoints(line);
    },
});
